autoPyTorch/data/base_feature_validator.py

```diff
@@ -152,4 +154,30 @@ def transform(
             np.ndarray:
                 The transformed array
         """
+
+        raise NotImplementedError()
+
+    def list_to_dataframe(
```

Comment:
That is a good question, in my pov I do not think it needs to be overall, however, if you check the implementation:
https://github.com/automl/Auto-PyTorch/blob/5f672b550d2fdc25c2a3865c20a78e44f04e796a/autoPyTorch/data/tabular_feature_validator.py#L394

it needs access to `self` to use the logger. 

I do not think it fits at the base_feature_validator, especially when dealing with non-tabular data, but I thought that would be out of the scope of this PR. We can raise an issue and get back to this in the future. I am sure it can be done way better.